torch/_fullgraph.py

```diff
@@ -0,0 +1,52 @@
+import dataclasses
+import logging
+import os
+from typing import Dict, List, Optional
+
+import torch
+import torch._inductor.package
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class _Compilation:
```

Comment:
Can you annotate these classes with what the API to use them will be, along with some example code?